go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -1473,3 +1480,29 @@ func DummyNextHopIPs() []net.IP {
 	}
 	return nextHops
 }
+
+// setGlobalForwardingRule either enables or disables forwarding for
+// all IPv4 and IPv6 interfaces by setting sysctl parameters based on
+// config.Gateway.DisableForwarding
+func setGlobalForwardingRule() error {
+	setForwardingSysctl := func(value string) error {
```

Comment:
So OVN-Kubernetes should not be setting global ip forwarding for anything. That is the role of the system admin of the node. It can cause undesired consequences if we globally enable forwarding for all interfaces on a server. So the stance of OVN-Kubernetes is we will set forwarding for the interfaces we manage.

The config is specific about this:
```
	&cli.BoolFlag{
		Name:        "disable-forwarding",
		Usage:       "Disable forwarding on OVNK controlled interfaces.",
		Destination: &cliConfig.Gateway.DisableForwarding,
	},
```

So I don't think you should set any global forwarding values. I also don't think you need to mess with the per interface forwarding regardless of what the forwarding mode is set to. It is required to allow some forwarding on OVNK managed interfaces for OVNK to function. Therefore the sysctl enablement is required no matter what. However, we can remove the restricted iptables rules when the flag is switched.

Reply:
@trozet While starting ovnkube-node we set[1] net.ipv4.ip_forward sysctl parameter to '1' when ipForwarding mode is set to 'Global' which in turn enables forwarding for all IPv4 interfaces. However we dont disable forwarding by setting  net.ipv4.ip_forward sysctl parameter to '0' when ipForwarding mode is switched to 'Restricted' from 'Global'. In newly created >4.14 cluster this bug does not impact as default ipForwarding mode is Restricted and net.ipv4.ip_forward sysctl parameter gets set to '0' by default. 

I do agree that we should not set IPv4 IP forwarding for all interfaces at ovn-kubernetes level. If ipForwarding mode is set to 'Restricted' then we can set net.ipv4.ip_forward sysctl parameter to '0' while starting[2] ovnkube-node itself. I can open a CNO PR additionally if you think that is ideal.

Addition and deletion of iptables rules can be taken care at ovn-kubernetes level as you agreed.

[1] - https://github.com/openshift/cluster-network-operator/blob/master/bindata/network/ovn-kubernetes/common/008-script-lib.yaml#L568
[2] - https://github.com/openshift/cluster-network-operator/blob/master/bindata/network/ovn-kubernetes/common/008-script-lib.yaml#L570

Reply:
Right. That is an OpenShift specific thing that should be done in CNO where you linked.

Reply:
I have raised a CNO PR as per agreement. https://github.com/openshift/cluster-network-operator/pull/2385
Thanks Tim & Surya :)